Aleksandra_Pyvovar/js/scripts.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Created by John Doe on 17.10.2017.
+ */
+document.onreadystatechange = function () {
+  if (document.readyState === 'interactive') {
+    initCalculator();
+  }
+};
+
+function initCalculator() {
+  var
+    calculator = document.getElementById('calculator'),
+    first = document.getElementById('first-number'),
+    second = document.getElementById('second-number'),
+    action = document.getElementById('action');
+
+
+  calculator.onsubmit = function(event) {
+    var
+      firstNumber = parseFloat(first.value),
+      secondNumber = parseFloat(second.value),
+      actionType = action.value.trim(),
+      result = calculate(firstNumber, secondNumber, actionType);
+
+    event.preventDefault();
+
+    if ((result !== undefined) && (!isNaN(result))  ) {
```

Comment:
Operators `!` and `!==` has a higher priority than `&&`, and You can omit `( ... )` brackets.
And write more simply:
```javascript
if (result !== undefined && !isNaN(result)) {
// ...
}
```
Or write an inversion of conditions:
```javascript
if (result === undefined && isNaN(result)) return;

resultRender(firstNumber, secondNumber, actionType, result);
this.reset();
```

Reply:
if (result === undefined && isNaN(result)) return;

Reply:
Yes, I forgot inverse `&&`, sorry.
```js
if (result === undefined || isNaN(result)) return;

resultRender(firstNumber, secondNumber, actionType, result);
this.reset();
```

Reply:
ah, yes. now I understand my problem))